src/clj/witan/send/model/prepare.clj

```diff
@@ -189,6 +176,63 @@
             (assoc coll [academic-year (states/join-need-setting need setting)] population))
           {} send-data))
 
+(defn get-filter-params [filter-what]
+  (into {} (filter #(every? (complement nil?) (vals (val %))) filter-what)))
+
+(defn build-pred [m]
+  (partial (:op (val m)) (:val (val m))))
+
+(def operators {:= =
+                :< <
+                :<= <=
+                :> >
+                :>= >=})
+
+(defn operator-key [filter-k]
+  (filter-k operators))
+
+(defn build-filter [filter-kv]
```

Comment:
I'd call this `remove-transitions-xf` as it returns a transducer function and is doing a remove rather than a filter. I'd also call `filter-kv` `remove-operation` as that is what it is defining and isn't *really* a kv at this point but a tuple.